src/server_manager/electron_app/windows/sign_windows_executable.mjs

```diff
@@ -0,0 +1,211 @@
+  args.push('--keystore', pfxCert);
+}
+
+function appendDigicertUsbJsignArgs(args, options) {
+  // extended validation certificate stored in USB drive
+  args.push('--storetype', 'PKCS11');
+
+  const subject = getOptionValue(options, 'subject', 'WINDOWS_SIGNING_EV_CERT_SUBJECT', false);
+  if (subject) {
+    args.push('--alias', subject);
+  }
+
+  var eTokenCfg;
+  switch (process.platform) {
+    case 'win32':
+      eTokenCfg = resolve(currentDirname(), 'digicert-usb-config', 'eToken-windows.cfg');
+      break;
+    case 'darwin':
+      eTokenCfg = resolve(currentDirname(), 'digicert-usb-config', 'eToken-macos.cfg');
+      break;
+    default:
+      throw new Error(`we do not support ev signing on ${process.platform}`);
+  }
+  args.push('--keystore', eTokenCfg);
+}
+
+/**
+ * Run jsign.jar according to the corresponding options targeting fileToSign.
+ * @param {string} fileToSign The path string of a file to be signed.
+ * @param {string[]} options The options to be passed to jsign. see https://ebourg.github.io/jsign/
+ * @returns {Promise<number>} A promise containing the exit code of jsign.
+ */
+function jsign(fileToSign, options) {
+  if (!options) {
+    throw new Error('options are required by jsign');
+  }
+  if (!fileToSign) {
+    throw new Error('fileToSign is required by jsign');
+  }
+
+  const jSignJarPath = resolve(outlineDirname(), 'third_party', 'jsign', 'jsign-4.0.jar');
+  const jsignProc = spawn('java', ['-jar', jSignJarPath, ...options, fileToSign], {
+    stdio: 'inherit',
+  });
+  return new Promise((resolve, reject) => {
+    jsignProc.on('error', reject);
+    jsignProc.on('exit', resolve);
+  });
+}
+
+/**
+ * Sign the target exeFile using a specific algorithm and options.
+ * @param {string} exeFile the full path of the exe file to be signed.
+ * @param {'sha1'|'sha256'} algorithm the algorithm used for signing.
+ * @param {object} options additional options (cli arguments) for signing.
+ *                         the options will also be read from environment
+ *                         variables.
+ */
+export async function signWindowsExecutable(exeFile, algorithm, options) {
+  const type = getOptionValue(options, 'certtype', 'WINDOWS_SIGNING_CERT_TYPE', false);
+  if (!type || type === 'none') {
+    console.info(`skip signing "${exeFile}"`);
+    return;
+  }
+
+  assert(!!exeFile, 'executable path is required');
+  assert(algorithm === 'sha1' || algorithm === 'sha256', 'hashing algorithm must be either "sha1" or "sha256"');
+
+  exeFile = resolve(exeFile);
+  await assertFileExists(exeFile, 'executable file "%s" does not exist');
+
+  const password = getOptionValue(options, 'password', 'WINDOWS_SIGNING_CERT_PASSWORD', true);
+
+  const jsignArgs = [
+    '--alg',
+    algorithm === 'sha256' ? 'SHA-256' : 'SHA-1',
+    '--tsaurl',
+    'http://timestamp.digicert.com',
+    '--storepass',
+    password,
+  ];
+
+  switch (type) {
```

Comment:
Do we need the two types? Which one do we use?

Reply:
We are using `digicert-usb`.

I used `pfx` to debug jsign without damaging the actual USB stick, it is a self-signed development certificate. We can use it to sign some internal (local) builds.